src/measure_theory/interval_integral.lean

```diff
@@ -647,8 +681,141 @@ begin
+
+lemma congr_ae {f g : α → E} (h : ∀ᵐ x ∂μ, x ∈ Ι a b → f x = g x) :
+  ∫ (x : α) in a..b, f x ∂μ = ∫ (x : α) in a..b, g x ∂μ :=
+interval_integral.congr_ae' (ae_interval_oc_iff.mp h).1 (ae_interval_oc_iff.mp h).2
+
+lemma integral_zero_ae  {f : α → E} (h : ∀ᵐ x ∂μ, x ∈ Ι a b → f x = 0) :
+  ∫ (x : α) in a..b, f x ∂μ = 0 :=
+calc ∫ x in a..b, f x ∂μ = ∫ x in a..b, 0 ∂μ : interval_integral.congr_ae h
+                     ... = 0                 : interval_integral.integral_zero
+
+lemma integral_indicator {a₁ a₂ a₃ : α} (h : a₂ ∈ Ioc a₁ a₃) {f : α → E} :
+  ∫ x in a₁..a₃, indicator {x | x ≤ a₂} f x ∂ μ = ∫ x in a₁..a₂, f x ∂ μ :=
+begin
+  have : {x | x ≤ a₂} ∩ Ioc a₁ a₃ = Ioc a₁ a₂, from Iic_inter_Ioc_of_le h.2,
+  rw [interval_integral.integral_of_le h.1.le, interval_integral.integral_of_le (h.1.le.trans h.2),
+      integral_indicator, measure.restrict_restrict, this],
+  exact measurable_set_Iic,
+  all_goals { apply measurable_set_Iic},
+end
+
 end order_closed_topology
 
+section continuity_wrt_parameter
+open topological_space
+
+variables {X : Type*} [topological_space X] [first_countable_topology X]
+variables {μ : measure α}
+
+lemma continuous_at_of_dominated_interval
+  {F : X → α → E} {x₀ : X} {bound : α → ℝ} {a b : α}
+  (hF_meas : ∀ᶠ x in 𝓝 x₀, ae_measurable (F x) (μ.restrict $ Ι a b))
+  (h_bound : ∀ᶠ x in 𝓝 x₀, ∀ᵐ t ∂(μ.restrict $ Ι a b), ∥F x t∥ ≤ bound t)
+  (bound_integrable : interval_integrable bound μ a b)
+  (h_cont : ∀ᵐ t ∂(μ.restrict $ Ι a b), continuous_at (λ x, F x t) x₀) :
+  continuous_at (λn, ∫ t in a..b, F n t ∂μ) x₀ :=
+begin
+  have gc := first_countable_topology.nhds_generated_countable x₀,
+  cases bound_integrable,
+  cases le_or_gt a b with hab hab;
+  [{ rw interval_oc_of_le hab at *,
+     simp_rw interval_integral.integral_of_le hab },
+   { rw interval_oc_of_lt hab at *,
+     simp_rw interval_integral.integral_of_ge (le_of_lt hab),
+     refine tendsto.neg _ }];
+  apply tendsto_integral_filter_of_dominated_convergence bound gc hF_meas
+        (mem_of_mem_nhds hF_meas : _) h_bound,
+  exacts [bound_integrable_left, h_cont, bound_integrable_right, h_cont]
+end
+
+lemma continuous_of_dominated_interval {F : X → α → E} {bound : α → ℝ} {a b : α}
+  (hF_meas : ∀ x, ae_measurable (F x) $ μ.restrict $ Ι a b)
+  (h_bound : ∀ x, ∀ᵐ t ∂(μ.restrict $ Ι a b), ∥F x t∥ ≤ bound t)
+  (bound_integrable : interval_integrable bound μ a b)
+  (h_cont : ∀ᵐ t ∂(μ.restrict $ Ι a b), continuous (λ x, F x t)) :
+  continuous (λn, ∫ t in a..b, F n t ∂μ) :=
+continuous_iff_continuous_at.mpr (λ x₀, continuous_at_of_dominated_interval
+  (eventually_of_forall hF_meas) (eventually_of_forall h_bound) ‹_› $ h_cont.mono $
+  λ _, continuous.continuous_at)
+
+end continuity_wrt_parameter
+
+section continuous_primitive
+open topological_space
+
+variables [topological_space α] [order_topology α] [opens_measurable_space α]
+          [first_countable_topology α] {a b : α} {μ : measure α}
+
+lemma continuous_at_primitive {f : α → E}
```

Comment:
This does not cover the case where you know that `f` is integrable on `[a, b]` (and have no information outside of this interval) and you want to show that `z -> ∫ x in a .. z, f z ∂ μ` is continuous on `[a, b]`, right? I think this is an important enough situation that it is worth stating a theorem general enough to cover this. (Also, I will need this formulation to prove the general version of FTC, assuming only `f'` is integrable :-)

Reply:
Done